docs/architecture/adr-009-evidence-module.md

```diff
@@ -0,0 +1,221 @@
+ determine the validity of submitted misbehaviour
+- Support the ability through governance to modify slashing penalties of any
+evidence type
+- Querier implementation to support querying params, evidence types, params, and
+all submitted valid misbehaviour
+
+### Types
+
+First, we define the `Evidence` interface type. The `x/evidence` module may implement
+its own types that can be used by many chains (e.g. `CounterFactualEvidence`).
+In addition, other modules may implement their own `Evidence` types in a similar
+manner in which governance is extensible. It is important to note any concrete
+type implementing the `Evidence` interface may include arbitrary fields such as
+an infraction time. We want the `Evidence` type to remain as flexible as possible.
+
+However, when submitting evidence to the `x/evidence` module, it must be submitted
+as an `Infraction` which includes mandatory fields outlined below. The `Infraction`
+type must include the validator's consensus address, which should be known by the
+`x/slashing` module, the height at which the infraction occured and the validator's
+power at same height in which the infraction occured.
+
+```go
+type Evidence interface {
+  Route() string
+  Type() string
+  ValidateBasic() sdk.Error
+  String() string
+}
+
+type Infraction struct {
+  Evidence
+
+  ConsensusAddress    sdk.ConsAddress
+  InfractionHeight    int64
+  Power               int64
+}
+```
+
+### Routing & Handling
+
+Each `Evidence` type must map to a specific unique route and be registered with
+the `x/evidence` module. It accomplishes this through the `Router` implementation. 
+
+```go
+type Router interface {
+  AddRoute(r string, h Handler) (rtr Router)
+  HasRoute(r string) bool
+  GetRoute(path string) (h Handler)
+  Seal()
+}
+```
+
+Upon successful routing through the `x/evidence` module, the `Evidence` type
+is passed through a `Handler`. This `Handler` is responsible for executing all
+corresponding business logic necessary for verifying the evidence. If no error
+is returned, the `Evidence` is considered valid.
+
+```go
+type Handler func(ctx sdk.Context, evidence Evidence) sdk.Error
+```
+
+### Submission
+
+Assuming the `Evidence` is valid, the corresponding slashing penalty is invoked
+for the `Evidence`'s `Type`. Keep in mind the slashing penalty for any `Type` can
+be configured through governance.
+
+```go
+type MsgSubmitInfraction struct {
+  Infraction
+}
+
+func handleMsgSubmitInfraction(ctx sdk.Context, keeper Keeper, msg MsgSubmitEvidence) sdk.Result {
+  if err := keeper.SubmitInfraction(ctx, msg.Infraction); err != nil {
+    return err.Result()
+  }
+
+  // emit events...
+
+  return sdk.Result{
+    // ...
+  }
+}
+```
+
+The `x/evidence` module's keeper is responsible for matching the `Evidence` against
+the module's router. Upon success the validator is slashed and the infraction is
+persisted. In addition, the validator is jailed is the `Evidence` type is configured
```

Comment:
```suggestion
persisted. In addition, the validator is jailed if the `Evidence` type is configured
```

Reply:
This has since been removed from the ADR :)